pkg/crd/mockCRDClient_test.go

```diff
@@ -0,0 +1,177 @@
+// Code generated by MockGen. DO NOT EDIT.
+// Source: k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1 (interfaces: CustomResourceDefinitionInterface)
+
+// Package crd is a generated GoMock package.
+package crd
+
+import (
+	context "context"
+	reflect "reflect"
+
+	gomock "github.com/golang/mock/gomock"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	v10 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/apimachinery/pkg/types"
+	watch "k8s.io/apimachinery/pkg/watch"
+)
+
+// MockCustomResourceDefinitionInterface is a mock of CustomResourceDefinitionInterface interface.
+type MockCustomResourceDefinitionInterface struct {
```

Comment:
A general question: why use mocks here? Couldn't you create a fake implementation instead?

Reply:
The fake.SimpleClient does not allow me to track what methods were called and with what parameters, the same way mocks do. My first implementation used the fake mocks from K8s but I ended up having to add custom tracking code and the test were pretty hard to follow.